javascript/code-challenges/challenges-04.test.js

```diff
@@ -0,0 +1,375 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named addAnimal that takes in array of animals (strings) and some callback function.
+
+This function should first create a new array. Then iterate over the input array and modify each value based on the callback function provided.
+
+Push each updated animal string into the new array. Return the new array.
+
+HINT: Look at the tests to see how the callback functions are used.
+
+------------------------------------------------------------------------------------------------ */
+
+function upper(str) {
+  return str.toUpperCase();
+}
+
+function lower(str) {
+  return str.toLowerCase();
+}
+
+const updateAnimal = (arr, callback) => {
+  const newArr = [];
+  arr.map(item => newArr.push(callback(item)));
+  return newArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function called sortNames that takes an array of names and sorts them alphabetically. Capital letters should come before lowercase letters.
+
+For example: 'Cat' would come before 'apple'
+------------------------------------------------------------------------------------------------ */
+
+const sortNames = (arr) => {
+  return arr.sort();
+};
```

Comment:
Great work. Since you're just returning arr.sort(), you can always use the implicit return to shorten your code. It doesn't necessarily make anything more efficient, just bragging rights to say "I did it in one line"
```js
const sortNames = (arr) => arr.sort();
```